integration/test/sameclusterid/same_cluster_id_test.go

```diff
@@ -0,0 +1,26 @@
+// +build k8srequired
+
+package clusterstate
+
+import (
+	"context"
+	"testing"
+)
+
+// Test_Same_Cluster_ID makes sure we can create the cluster with the same ID
+// after previous one is deleted. This improves coverage of resources
+// idempotentency.
+func Test_Same_Cluster_ID(t *testing.T) {
+	var err error
+	ctx := context.Background()
+
+	err = setup.DeleteTenantCluster(ctx, config)
+	if err != nil {
+		t.Fatalf("expected %#v got %#v", nil, err)
+	}
+
+	err = setup.CreateTenantCluster(ctx, config)
+	if err != nil {
+		t.Fatalf("expected %#v got %#v", nil, err)
+	}
```

Comment:
The test is fairly simple. Just delete existing cluster and replace it with another with the same ID.

Reply:
In this particular case this just makes sure we can recreate KMS encryption key properly which was the issue. But I wouldn't be surprised if we detect other things running that test.

In general it tests if we delete resources to the level in which we can create cluster for identical CR again.